covquestions-editor-app/src/components/QuestionnaireExecution.tsx

```diff
@@ -94,38 +120,76 @@ export const QuestionnaireExecution: React.FC<QuestionnaireExecutionProps> = ({
   return doRerender ? (
     <></>
   ) : (
-    <div className={classes.root}>
-      <Grid container item xs={12} className={`${classes.paddingRight} grid-row`} justify="flex-end">
-        <Button onClick={restartQuestionnaire} variant="contained" color="secondary">
-          Restart Questionnaire
-        </Button>
-      </Grid>
-      {isJsonInvalid ? (
-        <Grid item xs={12} className={`${classes.paddingRight} grid-row`}>
-          <Alert severity="warning">Cannot load questionnaire. JSON is invalid!</Alert>
-        </Grid>
-      ) : null}
-      <Grid item xs={12} className={`${classes.paddingRight} grid-row ${classes.execution}`}>
+    <Grid container direction="column" justify="space-between" className={`${classes.padding} overflow-pass-through`}>
+      <Grid item className={`${classes.execution}`}>
+        <Typography className={classes.internalStateHeadline}>Questionnaire Preview</Typography>
+        {isJsonInvalid ? <Alert severity="warning">Cannot load questionnaire. JSON is invalid!</Alert> : null}
         {result === undefined && currentQuestion ? (
-          <QuestionComponent currentQuestion={currentQuestion} handleNextClick={handleNextClick} />
+          <Paper className={classes.root}>
+            Hi
+            <Grid container direction="column" alignItems="stretch">
+              <Grid item xs={12}>
+                <QuestionFormComponent
+                  currentQuestion={currentQuestion}
+                  onChange={setCurrentValue}
+                  value={currentValue}
+                />
+              </Grid>
+              {currentQuestion.details ? (
+                <Grid item xs={12}>
+                  <Grid item xs={12}>
+                    <Typography>Hint:</Typography>
+                  </Grid>
+                  <Grid item xs={12}>
+                    <Typography>{currentQuestion.details}</Typography>
+                  </Grid>
+                </Grid>
+              ) : undefined}
+              <Grid container item xs={12} justify="space-between">
+                <Grid item>
+                  {progress > 0 ? (
```

Comment:
How do we define progress? Showing the first question => 0. Showing the last question => (n-1)/n. Showing the results => 1 ?

Reply:
I would make this dependent on the answers: 
0 answers progress = 0
all answered progress = 1

Reply:
All answer is hard, because we can skip questions => no answer for those. Anyway, we should not need the progress on the result screen. This should be always 1.